affinity/cpp-20/d0796r1.md

```diff
@@ -167,55 +202,56 @@ There are some additional challenges which we have been investigating but are no
 
 The `execution_resource` class provides an abstraction over a system's hardware capable of memory allocation, execution of light weight exeution agents or both.
 
-### `execution_resource` constructors
+> [*Note:* The `execution_resource` is required to be implemented such that the underlying software abstraction is initialised on when the `execution_resource` is constructed, maintained through reference counting and cleaned up on destruction of the final reference. *--end note*]
```

Comment:
Do we really need to mandate an implementation detail like this? if your execution resource is just a thread id for example, I don't think you need reference counting

Reply:
Tbh this requirement was quite hard to word, so it could definitely be improved. The reason I put this is that, since we return `execution_resource`s by value, they need to be reference counted to guarantee that two `execution_resource`s which identify the same hardware resource reference the same internal implementation. For example `execution_context(resource).resource() == resource`. Though this gets into equality semantics which are not yet defining.